ui.js

```diff
@@ -28,24 +29,136 @@ export default class UI {
     });
     this.game.addEventListener(Events.NEW_GENERATION, () => {
       this._setGeneration(this.game.generation);
+      this._setPopulation();
+      this._setCellStats();
     });
   }
 
   _reset() {
     this._setGeneration(this.game.generation);
+    this._setPopulation();
+    this._setCellStats();
   }
 
-  _setGeneration(value) {
-    this.generationCounter.innerHTML = value;
+  _setGeneration(generation) {
+    this.generationCounterOutput.setCount(generation);
+  }
+
+  _setPopulation() {
+    this.playerPopCounterOutput.forEach((popCounter, index) => {
+      popCounter.setCount(this.game.players[index].armySize);
+    });
+  }
+
+  _setCellStats() {
+    for (let i = 0; i < this.cellStats.length; ++i) {
+      for (let j = 0; j < this.cellStats[i].length; ++j) {
+        const army = this.game.board.cells[i][j].army;
+        if (army) {
+          const colourBox = createColourBox(army.player.colour.toString());
+          this.cellStats[i][j].setLabel(colourBox);
+          this.cellStats[i][j].setCount(army.size);
+          continue;
+        }
+
+        const colourBox = createColourBox();
+        this.cellStats[i][j].setLabel(colourBox);
+        this.cellStats[i][j].setCount(0);
+      }
+    }
   }
 
   _pause() {
     if (this.game.isRunning) {
       this.game.pause();
     }
   }
+
+  _render() {
+    this._renderTitle();
+    const gameControls = this._createGameControls();
+    const stats = this._createStats();
+    const cellStats = this.createCellStats();
+    this.body.append(gameControls);
+    this.body.append(stats);
+    this.body.append(cellStats);
+  }
+  
+  _renderTitle() {
+    const title = document.createElement("div");
+    title.innerHTML = "Square Game";
+    title.className = "title";
+    this.body.prepend(title); 
+  }
+
+  _createGameControls() {
+    const gameControls = document.createElement("div");
+    this.playButton = createButton(PLAY_ICON);
+    this.stopButton = createButton(STOP_ICON);
+    this.stepButton = createButton(STEP_ICON);
+    this.pauseButton = createButton(PAUSE_ICON);
+    gameControls.append(this.stopButton);
+    gameControls.append(this.pauseButton);
+    gameControls.append(this.playButton);
+    gameControls.append(this.stepButton);
+    return gameControls;
+  }
+
+  _createStats() {
+    const stats = document.createElement("div");
+    this.generationCounterOutput = new CounterOutput("Generation");
+    this.playerPopCounterOutput = new Array(this.game.players.length);
```

Comment:
if you are constructing an array from another array, then use the `map` method of arrays